### PR TITLE
Add unit tests for jsonrpc transaction encoding

### PR DESCRIPTION
**Pull request by ferranbt**
# Description

This PR is a follow-up for #919. It includes unit tests for encoding the `jsonrpc.Transaction` object.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Review by vcastellm — APPROVED** [review]
`gofmt` and LGTM

File: jsonrpc/testsuite/block-with-txn-bodies.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            "s": "0x3",
+            "hash": "0x0200000000000000000000000000000000000000000000000000000000000000",
+            "from": "0x0300000000000000000000000000000000000000",
+            "blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "blockNumber": "0x1",
+            "transactionIndex": "0x2"
+        }
+    ],
+    "uncles": null
+}
```
Comment (vcastellm) [inline jsonrpc/testsuite/block-with-txn-bodies.json:39]: would be great to add the new line if it doesn't break the tests

**Review by Stefan-Ethernal — APPROVED** [review]

**Comment by Stefan-Ethernal** [comment]
> `gofmt` and LGTM

Fixed https://github.com/0xPolygon/polygon-edge/pull/923/commits/383f05ed445cc02e6bdeaedd31b0679d352838ad & https://github.com/0xPolygon/polygon-edge/pull/923/commits/d496079b10478ec35779eb0c8d3f3b0e2db49d59

File: jsonrpc/testsuite/block-with-txn-bodies.json (hunk 1 above)
Comment (ferranbt) [reply jsonrpc/testsuite/block-with-txn-bodies.json:39]: Cannot do it from my vscode. It removes the line.

**Review by b00ris — APPROVED** [review]

**Pull request merged**